### PR TITLE
D2 - Created surveyor form submission POST request handler when button is clicked

### PR DESCRIPTION
**Pull request by rafiddewan**

### CODE REVIEW

**Review by tejash-patel — APPROVED** [review]

File: src/main/resources/static/surveyor.js [hunk 1]
```diff
@@ -39,10 +39,210 @@ $(document).ready(function() {
...
+
+                        //Check if all the MCQ options are filled
+                        if (!option) {
+                            alert("One of the MCQ options is empty for question: " + questionText);
+                            error = true
+                            return
+                        }
+
+                        //Add to data structure
+                        optionsJSON[option] = count
```
Comment (tejash-patel) [inline src/main/resources/static/surveyor.js:151]: we dont need count. just need to set all int to 0. e.g.
{
  "options": {
    "first option": 0,
    "second option": 0,
    "third opton": 0
  },
  "question": "Question String"
}

File: src/main/resources/static/surveyor.js [hunk 2]
```diff
@@ -39,10 +39,219 @@ $(document).ready(function() {
...
+        questions.children().each(function () {
+            //Contains jsonData of the question
+            if(error !== true) {
+                error = addJSONQuestion(jsonQuestions, this)
+                console.log(error)
+            }
+        });
+
+        //Don't do the ajax call if there was an error with one of the questions
+        if (error === true) return
```
Comment (nickcoutts99) [inline src/main/resources/static/surveyor.js:80]: you arent going to get here, if there was an error you are returning after each one

File: src/main/resources/static/surveyor.js [hunk 3]
```diff
@@ -39,10 +39,219 @@ $(document).ready(function() {
...
+
+        //Get survey response in order to get the survey id to do post requests for the questions
+        const surveyResponse = await handlePostRequest("http://localhost:8080/admin/survey", surveyName, true)
```
Comment (tejash-patel) [inline src/main/resources/static/surveyor.js:83]: Localhost would not work in production. Need to use relative url path like '/admin'.
Same for other endpoints.

File: src/main/resources/static/surveyor.js [hunk 4]
```diff
@@ -39,10 +39,219 @@ $(document).ready(function() {
...
+                method: 'POST',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json',
+                },
+                body: jsonData
+            };
+            try {
+                const fetchResponse = await fetch(url, settings);
+                return await fetchResponse.json();
```
Comment (nickcoutts99) [inline src/main/resources/static/surveyor.js:250]: why is this function only returning things if return promise ===true? Also i dont even think we need this logic here. We can just use .ajax with .done and .fail

File: src/main/resources/static/surveyor.js (hunk 2 above)
Comment (nickcoutts99) [reply src/main/resources/static/surveyor.js:80]: ah actually nvm we do get an error if the question is wrong

File: src/main/resources/static/surveyor.js [hunk 5]
```diff
@@ -39,10 +39,219 @@ $(document).ready(function() {
...
+            return
+        }
+
+        surveyName = JSON.stringify({"name": surveyName});
+        let questions = $('div[id^="Questions"]');
+
+        //Handle empty questions
+        if (questions.is(':empty')) {
+            alert("ERROR: There must be atleast 1 question in the survey!")
+            error = true;
```
Comment (nickcoutts99) [inline src/main/resources/static/surveyor.js:67]: dont need this, we return right after

File: src/main/resources/static/surveyor.js [hunk 6]
```diff
@@ -39,10 +39,219 @@ $(document).ready(function() {
...
+        //Carry the error for nested function
+        let error = false;
+
+        //Extract surveyName
+        let surveyName = $('#surveyName').val();
+
+        //Check if survey name is present
+        if(!surveyName) {
+            alert("ERROR: Please specify the name of the survey");
+            error = true;
```
Comment (nickcoutts99) [inline src/main/resources/static/surveyor.js:57]: dont need this we return right after

File: src/main/resources/static/surveyor.js (hunk 3 above)
Comment (nickcoutts99) [reply src/main/resources/static/surveyor.js:83]: also if the survey POST comes back with an error we shouldnt comtinue to send questions when we know dont know which surveys ID

**Review by nickcoutts99 — APPROVED** [review]

**Pull request merged**